test/core/super_component/test_super_component.py

```diff
@@ -0,0 +1,192 @@
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.retrievers.in_memory import InMemoryBM25Retriever
+from haystack.dataclasses import GeneratedAnswer
+from haystack.document_stores.in_memory import InMemoryDocumentStore
+from haystack.document_stores.types import DuplicatePolicy
+from haystack.utils.auth import Secret
+from haystack.core.serialization import default_from_dict, default_to_dict
+
+from haystack_experimental import SuperComponent
+
+@pytest.fixture
+def mock_openai_generator(monkeypatch):
+    """Create a mock OpenAI Generator for testing."""
+
+    def mock_run(self, prompt: str, **kwargs):
+        return {"replies": ["This is a test response about capitals."]}
+
+    monkeypatch.setattr(OpenAIGenerator, "run", mock_run)
+    return OpenAIGenerator(api_key=Secret.from_token("test-key"))
+
+
+@pytest.fixture
+def documents():
+    """Create test documents for the document store."""
+    return [
+        Document(content="Paris is the capital of France."),
+        Document(content="Berlin is the capital of Germany."),
+        Document(content="Rome is the capital of Italy.")
+    ]
+
+
+@pytest.fixture
+def document_store(documents):
+    """Create and populate a test document store."""
+    store = InMemoryDocumentStore()
+    store.write_documents(documents, policy=DuplicatePolicy.OVERWRITE)
+    return store
+
+@pytest.fixture
+def rag_pipeline(document_store):
+    """Create a simple RAG pipeline."""
+    @component
+    class FakeGenerator:
+        @component.output_types(replies=List[str])
+        def run(self, prompt: str, **kwargs):
+            return {"replies": ["This is a test response about capitals."]}
+
+    pipeline = Pipeline()
+    pipeline.add_component("retriever", InMemoryBM25Retriever(document_store=document_store))
+    pipeline.add_component("prompt_builder",
+                           PromptBuilder(
+                               template="Given these documents: {{documents|join(', ',attribute='content')}} Answer: {{query}}"))
+    pipeline.add_component("llm", FakeGenerator())
+    pipeline.add_component("answer_builder", AnswerBuilder())
+    pipeline.add_component("joiner", DocumentJoiner())
+
+    pipeline.connect("retriever", "prompt_builder.documents")
+    pipeline.connect("prompt_builder", "llm")
+    pipeline.connect("llm.replies", "answer_builder.replies")
+    pipeline.connect("retriever.documents", "joiner.documents")
+
+    return pipeline
+
+
+class TestSuperComponent:
+
+    def test_explicit_mapping(self, rag_pipeline):
+        # Create wrapper with input/output mappings
+        input_mapping = {
+            "search_query": ["retriever.query", "prompt_builder.query", "answer_builder.query"]
+        }
+        output_mapping = {
+            "answer_builder.answers": "final_answers"
+        }
+
+        wrapper = SuperComponent(
+            pipeline=rag_pipeline,
+            input_mapping=input_mapping,
```

Comment:
Why can't I just pass the `data` dict that I would pass to `pipeline.run`? The idea is the same, but from a user's perspective, it's another syntax to learn.

Reply:
They can't pass a data dict because this is a component and not a pipeline. The mapping to input/output sockets is necessary so that the users can connect this component to other components.

However, users don't have to provide a mapping, auto-resolution of the mapping should work in most cases.